hub/api/dataset.py

```diff
@@ -32,18 +32,17 @@ def __init__(
             mode (str): Mode in which the dataset is opened.
                 Supported modes include ("r", "w", "a") plus an optional "+" suffix.
                 Defaults to "a".
-            ds_slice (slice): The slice object restricting the view
-                of this dataset's tensors. Defaults to slice(None, None, None).
-                Used internally for iteration.
             provider (StorageProvider, optional): The storage provider used to access
                 the data stored by this dataset.
+            index: The Index object restricting the view of this dataset's tensors.
```

Comment:
nit-pick: "The Index object is used to restrict the view of this dataset's tensors."